packages/sdk-components-react/src/html-embed.tsx

```diff
@@ -8,53 +8,110 @@ import {
 import { mergeRefs } from "@react-aria/utils";
 import { ReactSdkContext } from "@webstudio-is/react-sdk";
 
-type Props = {
-  code: string;
-  executeScriptOnCanvas?: boolean;
-  clientOnly?: boolean;
+const insertScript = (
+  sourceScript: HTMLScriptElement
+): Promise<HTMLScriptElement> => {
+  return new Promise((resolve, reject) => {
+    const script = document.createElement("script");
+
+    // Copy all attributes from the source script to the new script, because we are going to replace the source script with the new one
+    // and the user might rely on some attributes.
+    for (const { name, value } of sourceScript.attributes) {
+      script.setAttribute(name, value);
+    }
+
+    if (sourceScript.src) {
+      script.onload = () => {
+        resolve(script);
+      };
+      script.onerror = reject;
+      script.src = sourceScript.src;
+    } else {
+      script.textContent = sourceScript.innerText;
+    }
+
+    document.head.appendChild(script);
+    sourceScript.replaceWith(script);
+
+    // Run the callback immediately for inline scripts.
+    if (sourceScript.hasAttribute("src") === false) {
```

Comment:
why check property above and attribute here?

Reply:
where is the property check you are mentioning?

Reply:
oh you mean sourceScript.src ? no reason, fixing